etcd/src/main/scala/io/buoyant/etcd/Etcd.scala

```diff
@@ -0,0 +1,115 @@
+package io.buoyant.etcd
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.twitter.finagle.{Path, Service}
+import com.twitter.finagle.buoyant.FormParams
+import com.twitter.finagle.http.{MediaType, Message, Method, Request, Response, Status}
+import com.twitter.io.Buf
+import com.twitter.util._
+
+case class Version(etcdserver: String, etcdcluster: String)
+
+case class UnexpectedResponse(
+  method: Method,
+  uri: String,
+  params: Seq[(String, String)],
+  status: Status,
+  state: Etcd.State
+) extends Exception({
+  val ps = params.map { case (k, v) => s"($k -> $v)" }.mkString(", ")
+  s"""$method $uri [$ps] $status"""
+})
+
+object Etcd {
+
+  private[etcd] val keysPrefixPath = Path.Utf8("v2", "keys")
+  private[etcd] val versionPath = Path.Utf8("version")
```

Comment:
I will echo @adleong's comment on an earlier revision of this branch: it strikes me as weird to be using a Finagle `Path` to represent a URI, both because it's surprising semantically and because they have different formats (although in this case of course they map to the same string).